crates/core/src/transport/symmetric_message.rs

```diff
@@ -169,7 +175,7 @@ mod test {
         let key = Aes128Gcm::new(&[0; 16].into());
         let packet = SymmetricMessage::ack_error(&key)?;
 
-        let _packet = PacketData::<1000>::encrypt_symmetric(packet.data(), &key);
+        let _packet = PacketData::<Symmetric, 1000>::encrypt_symmetric(packet, &key);
```

Comment:
Shouldn't packet already be encrypted when returned from the ack_error() function?